pages/blog/socketio-part1.md

```diff
@@ -0,0 +1,64 @@
+
+In the spirit of not repeating the sins of the past (i.e. not focusing on the client<->server interface), I started looking into how one can document a Socket.IO API. By document, I mean putting together some sort of spec or contract that can be agreed upon and shared with the consumers of the API. To my surprise, I couldn’t find a lot of resources on the topic. Coming from the REST world (where [OpenAPI](https://www.openapis.org/) is nowadays the de facto standard) and from the [GraphQL](https://graphql.org/) world (where everything is strictly typed), I found myself rather disappointed. It feels like the industry has been treating the documentation of event-driven APIs, such as Socket.IO or WebSocket ones, as a niche area which is not worth standardising. This is odd given how popular event-driven architectures have become over the past decade.
+
+After getting over the initial shock of this realisation, I finally managed to put together some properly worded google searches and came across [AsyncAPI](https://www.asyncapi.com/). Apparently, there is a growing community out there which serves this exact purpose of documenting event-driven APIs! Since 2017, the AsyncAPI folks have been developing a protocol agnostic specification for asynchronous APIs (inspired by OpenAPI), along with tooling such as spec parsers and code generators. Diving deeper into my google search, I looked for articles and tutorials on how one may express a Socket.IO API using the AsyncAPI specification. The only single resource I managed to find on the public cyberspace was [this](https://stackoverflow.com/a/45701602/10975573) (not very comprehensive) StackOverflow answer. It was at this point when I realised that I was on my own, facing the problem of modelling the Socket.IO protocol using the AsyncAPI semantics. I decided to take on the challenge and if successful, write a blog post documenting my journey. So here we are.
```

Comment:
```suggestion
After getting over the initial shock of this realization, I finally managed to put together some properly worded google searches that led to the [AsyncAPI](https://www.asyncapi.com/) community. Apparently, there is a growing community out there that serves this exact purpose of documenting event-driven APIs! Since 2017, the AsyncAPI folks have been developing an agnostic protocol specification for asynchronous APIs (inspired by OpenAPI), along with tooling such as spec parsers and code generators. Diving deeper into my google search, I looked for articles and tutorials on how one may express a Socket.IO API using the AsyncAPI specification. The only resource I managed to find on the internet was [this](https://stackoverflow.com/a/45701602/10975573) (not very comprehensive) StackOverflow answer. At this point, I realized that I was on my own, facing the problem of modeling the Socket.IO protocol using the AsyncAPI semantics. I decided to take on the challenge and, if successful, write a blog post documenting my journey. So here we are.
```

Not sure why the differences are not showing, but I corrected English grammar and typos.

Reply:
I only applied a subset of your suggestions on this one. With regards to typos, I would like to keep the UK style of spelling so that it's coherent with the original article.

Regarding the `it was at this point...` and `public cyberspace`, I wrote these in a humorous manner and I'd also like to keep them, to maintain this flavour of silliness in the article.